craft_cli/completion/completion.py

```diff
@@ -215,7 +216,7 @@ def complete(shell_cmd: str, get_dispatcher: Callable[[], craft_cli.Dispatcher])
     command_map: Dict[str, CommandMapping] = {}
     for name, cmd_cls in dispatcher.commands.items():
         parser = argparse.ArgumentParser()
-        cmd = cmd_cls(None)
+        cmd = cmd_cls(app_config)
```

Comment:
Thanks for including me here - this was the piece where I wasn't understanding the need, as I thought the dispatcher had the app config. But refreshing based on the [craft-application implementation](https://github.com/canonical/craft-application/blob/657d1fe9497c0e9c5c2a4d237b25b9b666080eeb/craft_application/application.py#L623) it makes more sense.